instrumentation/elasticsearch/elasticsearch-api-client-8.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/elasticsearch/apiclient/v8_0/ElasticsearchEndpointMap.java

```diff
@@ -0,0 +1,774 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.elasticsearch.apiclient.v8_0;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ElasticsearchEndpointMap {
+  private static final Set<String> searchEndpoints =
+      Stream.of(
+              "search",
+              "async_search.submit",
+              "msearch",
+              "eql.search",
+              "terms_enum",
+              "search_template",
+              "msearch_template",
+              "render_search_template")
+          .collect(Collectors.toCollection(HashSet::new));
+
+  private static Map<String, String[]> routesMap;
+
+  private ElasticsearchEndpointMap() {}
+
+  public static boolean isSearchEndpoint(String endpointId) {
+    return searchEndpoints.contains(endpointId);
+  }
+
+  public static Map<String, String[]> get() {
+    if (routesMap == null) {
```

Comment:
this lazy initialization is not thread safe

Reply:
Good point. Fixed it.

Reply:
did you consider just initializing this map in the static initializer?

Reply:
Currently the only method that is called in this class is `get` so this map will always be initialized. If you later add more methods and wish to decouple the map initialization from these then you could move the map initialization to an inner class static field.

Reply:
will move it to a static initializer

Reply:
moved to static initializer